### PR TITLE
dual funding -> API changes/breakage

### PR DESCRIPTION
**Pull request by niftynei**
Built on #5650, updates to latest + greatest dual-funding spec proposal.

Awaiting interop with ACINQ/Eclair, and requires lnprototest updates (so in draft til these are updated).

Biggest change is that anchors aren't assumed for v2 opens now.

### CODE REVIEW

File: connectd/connectd.c [hunk 1]
```diff
@@ -305,8 +306,8 @@ struct io_plan *peer_connected(struct io_conn *conn,
 		status_peer_unusual(id, "Unsupported feature %u", unsup);
 		msg = towire_warningfmt(NULL, NULL, "Unsupported feature %u",
 					unsup);
-		msg = cryptomsg_encrypt_msg(tmpctx, cs, take(msg));
-		return io_write(conn, msg, tal_count(msg), io_close_cb, NULL);
+		msg = cryptomsg_encrypt_msg(NULL, cs, take(msg));
+		return io_write_wire(conn, take(msg), io_close_cb, NULL);
```
Comment (cdecker) [inline connectd/connectd.c:310]: Good catch, just verified that `io_write_wire` takes `data`, despite not being annotated as such. It uses `tal_dup_talarr` which is marked as `TAKES`

**Review by cdecker — COMMENTED** [review]
Very nice cleanups, looking forward to merging this. All I could find was a slight issue with the `ok =` chaining used to compute some `amount_sat` values. It can happen that we clobber an amount with an uninitialized value without noticing if I'm not mistaken.

File: common/features.c [hunk 2]
```diff
@@ -165,9 +165,9 @@ static const struct dependency feature_deps[] = {
 	/* BOLT-f53ca2301232db780843e894f55d95d512f297f9 #9:
 	 * Name                | Description  | Context  | Dependencies  |
 	 * ...
-	 * `option_dual_fund`  | ...          | ...      | `option_anchor_outputs`
+	 * `option_dual_fund`  | ...          | ...      | `option_static_remotekey`
 	 */
-	{ OPT_DUAL_FUND, OPT_ANCHOR_OUTPUTS },
```
[the remark this answers is not in the document]
Comment (niftynei) [reply common/features.c:170]: We've removed the implicit dependence on anchor_outputs, instead we just assume that static_remotekey is required.

**Comment by niftynei** [comment]
test failure unrelated to changes:

```
Valgrind error file: valgrind-errors.24405
==24405== Conditional jump or move depends on uninitialised value(s)
==24405==    at 0x49E4B56: __vfprintf_internal (vfprintf-internal.c:1516)
==24405==    by 0x49F6519: __vsnprintf_internal (vsnprintf.c:114)
==24405==    by 0x1EC4CB: do_vfmt (str.c:66)
==24405==    by 0x1EC5D8: tal_vfmt_ (str.c:92)
==24405==    by 0x11A356: paymod_log (libplugin-pay.c:167)
==24405==    by 0x11B4D2: payment_chanhints_apply_route (libplugin-pay.c:534)
==24405==    by 0x11E9B9: payment_compute_onion_payloads (libplugin-pay.c:1707)
==24405==    by 0x11FF6C: payment_continue (libplugin-pay.c:2135)
==24405==    by 0x1245E0: adaptive_splitter_cb (libplugin-pay.c:3800)
==24405==    by 0x11FF13: payment_continue (libplugin-pay.c:2123)
==24405==    by 0x12061E: retry_step_cb (libplugin-pay.c:2301)
==24405==    by 0x11FF13: payment_continue (libplugin-pay.c:2123)
==24405== 
```

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/62af76d930fcd56a314a96b557f5795613b3b495

**Pull request merged**